layers/gpu_validation/gpu_validation.cpp

```diff
@@ -1232,94 +1232,19 @@ void gpuav_state::CommandBuffer::Process(VkQueue queue) {
-                if (buffer_view == VK_NULL_HANDLE) {
-                    data[index++] = UINT_MAX;
-                } else {
-                    auto buffer_view_state = desc.GetBufferViewState();
-                    data[index++] = static_cast<uint32_t>(buffer_view_state->buffer_state->createInfo.size);
-                }
-            }
-            break;
-        }
-        case cvdescriptorset::DescriptorClass::Mutable: {
-            auto mutable_binding = static_cast<const cvdescriptorset::MutableBinding *>(binding);
-            for (uint32_t di = 0; di < mutable_binding->count; di++) {
-                const auto &desc = mutable_binding->descriptors[di];
-                if (!mutable_binding->updated[di]) {
-                    data[index++] = 0;
-                    continue;
-                }
-                switch (desc.ActiveType()) {
-                    case VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER:
-                    case VK_DESCRIPTOR_TYPE_STORAGE_BUFFER:
-                    case VK_DESCRIPTOR_TYPE_UNIFORM_TEXEL_BUFFER:
-                    case VK_DESCRIPTOR_TYPE_STORAGE_TEXEL_BUFFER:
-                        data[index++] = static_cast<uint32_t>(desc.GetBufferSize());
-                        break;
-                    default:
-                        data[index++] = 1;
-                        break;
-                }
-            }
-            break;
-        }
-        default: {
-            for (uint32_t i = 0; i < binding->count; i++, index++) {
-                data[index] = static_cast<uint32_t>(binding->updated[i]);
-            }
-            break;
-        }
-    }
-}
-
 // For the given command buffer, map its debug data buffers and update the status of any update after bind descriptors
 void GpuAssisted::UpdateInstrumentationBuffer(gpuav_state::CommandBuffer *cb_node) {
-    uint32_t *data;
-    for (const auto &buffer_info : cb_node->di_input_buffer_list) {
-        for (const auto &set_buffer : buffer_info.descriptor_set_buffers) {
-            if (set_buffer.update_at_submit.size() > 0) {
-                VkResult result = vmaMapMemory(vmaAllocator, set_buffer.allocation, reinterpret_cast<void **>(&data));
-                if (result == VK_SUCCESS) {
-                    for (const auto &update : set_buffer.update_at_submit) {
-                        SetBindingState(data, update.first, update.second);
-                    }
-                    // Flush the descriptor state buffer before unmapping so that the new state is visible to the GPU
-                    result = vmaFlushAllocation(vmaAllocator, set_buffer.allocation, 0, VK_WHOLE_SIZE);
-                    // No good way to handle this error, we should still try to unmap.
-                    assert(result == VK_SUCCESS);
-                    vmaUnmapMemory(vmaAllocator, set_buffer.allocation);
-                }
+    for (auto &buffer_info : cb_node->di_input_buffer_list) {
+        uint64_t *address_data_ptr{nullptr};
```

Comment:
Complete nitpick I just think it helps readability.

```suggestion
        VkDeviceAddress *address_data_ptr{nullptr};
```